lib/StackDetector/BaseComposerTypeDetector.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Einenlum\PhpStackDetector\StackDetector;
+
+use Einenlum\PhpStackDetector\Composer\PackageVersionProvider;
+use Einenlum\PhpStackDetector\Stack;
+use Einenlum\PhpStackDetector\StackType;
+
+abstract class BaseComposerTypeDetector
```

Comment:
I tend to use traits and interfaces rather than relying on abstract classes which provides better composition, but if it's required I'm okay with it too :)